azuread/resource_domain_service.go

```diff
@@ -0,0 +1,492 @@
+package azuread
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/domainservices/mgmt/2017-06-01/aad"
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-09-01/network"
+	"github.com/hashicorp/go-azure-helpers/response"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azuread/azuread/helpers/ar"
+	"github.com/terraform-providers/terraform-provider-azuread/azuread/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azuread/azuread/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azuread/azuread/helpers/validate"
+)
+
+func resourceDomainService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDomainServiceCreate,
+		Read:   resourceArmDomainServiceRead,
+		Update: resourceArmDomainServiceUpdate,
+		Delete: resourceArmDomainServiceDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"domain_security_settings": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"ntlm_v1": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(aad.NtlmV1Enabled),
+								string(aad.NtlmV1Disabled),
+							}, false),
+							Default: string(aad.NtlmV1Enabled),
+						},
+						"sync_ntlm_passwords": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(aad.SyncNtlmPasswordsEnabled),
+								string(aad.SyncNtlmPasswordsDisabled),
+							}, false),
+							Default: string(aad.SyncNtlmPasswordsEnabled),
+						},
+						"tls_v1": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(aad.TLSV1Enabled),
+								string(aad.TLSV1Disabled),
+							}, false),
+							Default: string(aad.TLSV1Enabled),
+						},
+					},
+				},
+			},
+
+			"filtered_sync": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(aad.FilteredSyncEnabled),
+					string(aad.FilteredSyncDisabled),
+				}, false),
+				Default: string(aad.FilteredSyncEnabled),
+			},
+
+			"ldaps_settings": {
```

Comment:
settings is implied so:
```suggestion
			"ldaps": {
```